src/Workspaces/Remote/Core/ExportProviderBuilder.cs

```diff
@@ -198,7 +198,7 @@ private void ThrowOnUnexpectedErrors(CompositionConfiguration configuration, Com
             {
                 // The ToString for the composition failed exception doesn't output a nice set of errors by default, so log it separately
                 LogError($"Encountered errors in the MEF composition:{Environment.NewLine}{ex.ErrorsAsString}");
-                throw;
+                throw new CompositionFailedException(ex.Message + Environment.NewLine + ex.ErrorsAsString);
```

Comment:
The `RemoteInitializationService.InitializeAsync` API needs some work - it returns error message that we log in-proc, but when the composition fails we throw an exception here and the message is never returned. 

Also, the error message is not shown in the gold bar. We only log it.

Updating the exception message here with the composition failure details makes it easy to see in the gold bar what wen wrong. Perhaps we can refactor this to not throw (or catch the exception in InitializeAsync) and display message in the gold bar.